File: lib/classes/update_status_info.py
from typing import List
from lib.classes.addresses.ipv4 import IPv4Addr
from lib.classes.endpoint.endpoint_value import EndpointValue


class UpdateStatusInfo:
    def __init__(self, is_ack: bool, endpoint_id: int, local_ip_address: IPv4Addr, endpoint_values: List[EndpointValue]):
        self.is_ack = is_ack
        self.endpoint_id = endpoint_id
        self.local_ip_address = local_ip_address
        self.endpoint_values = endpoint_values

    def is_ack(self):
        return self.is_ack

    def get_endpoint_id(self):
        return self.endpoint_id

    def get_local_ip_address(self):
        return self.local_ip_address

    def get_endpoint_values(self):
        return self.endpoint_values

    def set_is_ack(self, is_ack: bool):
        self.is_ack = is_ack

    def set_endpoint_id(self, endpoint_id: int):
        self.endpoint_id = endpoint_id
    
    def set_local_ip_address(self, local_ip_address: IPv4Addr):
        self.local_ip_address = local_ip_address
    
    def set_endpoint_values(self, endpoint_values: List[EndpointValue]):
        self.endpoint_values = endpoint_values
    

File: HBRS-Rust/hbrs/src/classes/update_status_info.rs
use super::{addresses::ipv4::IPv4Addr, endpoint::endpoint_value::EndpointValue};
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
#[derive(Debug, Clone)]
pub struct UpdateStatusInfo {
    is_ack: bool,
    endpoint_id: u64,
    local_ip_address: IPv4Addr,
    endpoint_values: Vec<EndpointValue>,
}

impl UpdateStatusInfo{
    pub fn new(is_ack: bool, endpoint_id: u64, local_ip_address: IPv4Addr, endpoint_values: Vec<EndpointValue>) -> Self{
        Self{
            is_ack,
            endpoint_id,
            local_ip_address,
            endpoint_values,
        }
    }

    pub fn get_is_ack_ref(&self) -> &bool {
        &self.is_ack
    }
    
    pub fn get_is_ack_mut(&mut self) -> &mut bool {
        &mut self.is_ack
    }
    
    pub fn get_endpoint_id_ref(&self) -> &u64 {
        &self.endpoint_id
    }
    
    pub fn get_endpoint_id_mut(&mut self) -> &mut u64 {
        &mut self.endpoint_id
    }
    
    pub fn get_local_ip_address_ref(&self) -> &IPv4Addr {
        &self.local_ip_address
    }
    
    pub fn get_local_ip_address_mut(&mut self) -> &mut IPv4Addr {
        &mut self.local_ip_address
    }
    
    pub fn get_endpoint_values_ref(&self) -> &Vec<EndpointValue> {
        &self.endpoint_values
    }
    
    pub fn get_endpoint_values_mut(&mut self) -> &mut Vec<EndpointValue> {
        &mut self.endpoint_values
    }
}